docs/spec/concepts.rst

```diff
@@ -18,17 +280,34 @@ Example::
           e = [e]
       ...
 
-A type factored by ``T1 | T2 | ...`` is a supertype
-of all types ``T1``, ``T2``, etc., so that a value that
-is a member of one of these types is acceptable for an argument
-annotated by ``T1 | T2 | ...``.
+A union type ``T1 | T2``, where ``T1`` and ``T2`` are fully static types,
+represents the set of values formed by the union of the sets of values
+represented by ``T1`` and ``T2``. Thus, by the definition of the supertype
+relation, the union ``T1 | T2`` is a supertype of both ``T1`` and ``T2``. Via
+materialization, the gradual types ``S1`` and ``S2`` are both assignable to a
+gradual union type ``S1 | S2``.
```

Comment:
This sentence doesn't make much sense to me. It might be clearer if you say more explicitly what S1 | S2 means if the two types are not fully static.

Reply:
Does this added sentence help to clarify?

> A gradual union type ``S1 | S2``, where ``S1`` and ``S2`` are gradual types,
represents all possible sets of values that could be formed by union of the
possible sets of values represented by materializations of ``S1`` and ``S2``.

Reply:
Although actually I think this explanation might be more directly related to the sentence you asked about:

> Via materialization, the gradual types ``S1`` and ``S2`` are both assignable to
a gradual union type ``S1 | S2``. For any materialization of ``S1`` to ``T1``
and ``S2`` to ``T2``, ``S1 | S2`` can likewise be materialized to ``T1 | T2``,
where ``T1`` and ``T2`` are subtypes of ``T1 | T2``.

Reply:
The current version of this section reads much better, yes. Thanks!